### DISCUSSION TITLE
Executing the function computeproperty() in my own Material object only once at the start time.

### DISCUSSION THREAD

**Discussion by 6ImmortalFish6** [idaholab/moose#24194] [Q&A General] [answered]
I'm wondering how can I  make my custom material object run the computeProperty() function only once after the program starts. For example, if I am conducting transient analysis with a start time of 0s, end time of 10s, and a time step dt of 1s, computeProperty() should be executed every dt during the simulation. How can I modify it to only execute at the beginning and not during any subsequent iterations?

I came up with a solution, but I encountered a problem in the process. I plan to add an "if" statement that uses the initial time value as a criterion. If the current time is different from the initial time, then just skip it. However, I am uncertain about how to retrieve the initial time value in the Material System. Could you please advise me on how to obtain the initial time value?

Could you please advise me on how to solve the first problem and second problem? 

Thank you very much for your help.

**Comment by GiudGiud** [comment]
Hello

You can set `compute = False` in a material.
It will then become your responsibility to call the compute routines on the material, from wherever you need it computed

Guillaume

**Comment by hugary1995** [comment]
You have access to `_t_step` which is the current time step number.

**Reply by GiudGiud** [reply, marked as answer]
There's some examples of materials in tensor_mechanics and phase field that use _t_step
such as 
tensor_mechanics/src/materials/LinearViscoelasticityBase
